stencils/pace/stencils/testing/test_translate.py

```diff
@@ -167,6 +168,46 @@ def process_override(threshold_overrides, testobj, test_name, backend):
             )
 
 
+def perturb(input):
+    roundoff = 1e-16
+    for data in input.values():
+        if isinstance(data, np.ndarray) and data.dtype in (np.float64, np.float32):
+            not_fill_value = data < 1e10
+            # multiply data by roundoff-level error
+            data[not_fill_value] *= 1.0 + np.random.uniform(
+                low=-roundoff, high=roundoff, size=data[not_fill_value].shape
+            )
+
+
+N_THRESHOLD_SAMPLES = 10
```

Comment:
Did you try a couple of numbers and the threshold stopped ~changing after a few?

Reply:
I just put 10 and called it good. There's a tradeoff to make between statistical robustness and performance. I'll experiment with this a bit and see how it changes, but I doubt I can use much smaller than this, and we don't need to use a larger one if we use the right scale factor (we probably should set threshold a bit larger than the greatest difference we observe). In these cases I set the threshold about 2-5x larger than the "suggested" (so 5e-10 instead of 1e-10, for example).

Reply:
I tested some more values on Riem_Solver3, but I think I'll just leave it at 10. Even at 20 runs, it wasn't enough to really nail down a max relative difference for `w`, but 10 was enough to have consistent output differences for every other value.